src/filesystem/files.rs

```diff
@@ -128,6 +128,14 @@ where
     pub fn flush(&mut self) -> Result<(), Error<D::Error>> {
         self.volume_mgr.flush_file(self.raw_file)
     }
+
+    /// Consume the `File` handle and close it. The behavior of this is similar
+    /// to using [`core::mem::drop`] or letting the `File` go out of scope,
+    /// except this lets the user handle any errors that may occur in the process,
+    /// whereas when using drop, any errors will be discarded silently.
+    pub fn close(self) -> Result<(), Error<D::Error>> {
+        self.volume_mgr.close_file(self.raw_file)
+    }
```

Comment:
I think you need to forget the File here, to prevent the Drop trait from running. Otherwise it'll try and close it twice, and fail the second time - which just seems wasteful.